### PR TITLE
fix: Experiment table, right-click context menu [WEB-1942]

### PR DESCRIPTION
**Pull request by mapmeld**
## Description

The bug: Options in the right-click context menu sometimes close the menu but do not execute the action (archive/unarchive, open edit modal, etc.)

A few days after the bug report, we upgraded to glide-table v6. The bug is still present.
The easiest way to repeat the bug is to right-click the top row, and then click an option which is overlayed over the table header, or outside the glide table.

Screen Shot 2024-01-25 at 1 03 55 PM

The issue was `useOutsideClickHandler` - from https://github.com/determined-ai/determined/pull/6532 - we use it to close the ContextMenu by clicking somewhere outside. We keep this handler but make it trigger in fewer cases.

I fixed an unrelated bug with the "Experiment updated successfully" message. `notification.success` (pictured) is replaced by more standard `openToast`.

Screen Shot 2024-01-25 at 10 03 40 AM

## Test Plan

It is easier to test context menu overlap on `/det/projects/107/experiments`

- Right-click an experiment row to open the context menu. Right-click another row to open a new menu. Click somewhere away from the menu to close. 

- Right-click a row, select Edit, and change the title or description of the experiment. A confirmation message appears.

- Right-click the top row, and then click an option (Archive/Edit) which is above the table header, or outside the glide table
- Right-click a row where we have checkbox, a link, text, blank space, etc. and select Edit - a modal should appear each time

## Checklist

- [x] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: webui/react/src/pages/F_ExpList/F_ExperimentList.tsx [hunk 1]
```diff
@@ -551,7 +552,7 @@ const F_ExperimentList: React.FC<Props> = ({ project }) => {
           break;
         case ExperimentAction.Edit:
           if (data) updateExperiment(data);
-          notification.success({ message: 'Experiment updated successfully' });
+          openToast({ severity: 'Confirm', title: 'Experiment updated successfully' });
```
Comment (keita-determined) [inline webui/react/src/pages/F_ExpList/F_ExperimentList.tsx:555]: whats the difference between notification and openToast?

File: webui/react/src/pages/F_ExpList/glide-table/contextMenu.tsx [hunk 2]
```diff
@@ -16,9 +16,7 @@ function useOutsideClickHandler(ref: MutableRefObject<any>, handler: (event: Eve
       if (
         ref.current &&
         !ref.current.contains(event.target) &&
-        !(event.target ? (event.target as Element) : null)?.classList?.contains(
-          'ant-dropdown-menu-title-content',
-        )
+        !(event.srcElement as Element).className?.includes('ant-dropdown')
```
Comment (keita-determined) [inline webui/react/src/pages/F_ExpList/glide-table/contextMenu.tsx:19]: `srcElement` is marked as  deprecated. Can we change it to something else?

File: webui/react/src/pages/F_ExpList/F_ExperimentList.tsx (hunk 1 above)
Comment (mapmeld) [reply webui/react/src/pages/F_ExpList/F_ExperimentList.tsx:555]: Via GQ: "openToast is a style wrapper over notification.open  and we should prefer using openToast"

File: webui/react/src/pages/F_ExpList/glide-table/contextMenu.tsx (hunk 2 above)
Comment (mapmeld) [reply webui/react/src/pages/F_ExpList/glide-table/contextMenu.tsx]: ok it looks like I should continue using `event.target as Element` - this makes the change small, but I continue to see positive results

File: webui/react/src/pages/F_ExpList/glide-table/contextMenu.tsx [hunk 3]
```diff
@@ -16,9 +16,7 @@ function useOutsideClickHandler(ref: MutableRefObject<any>, handler: (event: Eve
       if (
         ref.current &&
         !ref.current.contains(event.target) &&
-        !(event.target ? (event.target as Element) : null)?.classList?.contains(
-          'ant-dropdown-menu-title-content',
-        )
+        !(event.target as Element).className?.includes('ant-dropdown')
```
Comment (keita-determined) [inline webui/react/src/pages/F_ExpList/glide-table/contextMenu.tsx:19]:
```suggestion
        event.target instanceof Element &&
        !event.target.className?.includes('ant-dropdown')
```

`target` does not always inherit `Element`, so i think the above code is safer

**Review by keita-determined — APPROVED** [review]
Left a comment, otherwise LGTM

File: webui/react/src/pages/F_ExpList/glide-table/contextMenu.tsx (hunk 3 above)
Comment (mapmeld) [reply webui/react/src/pages/F_ExpList/glide-table/contextMenu.tsx]: ok I like the use of `instanceof` but have a different version of the condition

**Pull request merged**